src/numbers/NumberWithProperties.java

```diff
@@ -0,0 +1,98 @@
+package numbers;
+
+import java.util.function.Supplier;
+import java.util.stream.IntStream;
+
+class NumberWithProperties {
+
+    private final long number;
+    private final String[] properties;
+
+    public NumberWithProperties(final long number) {
+        this.number = number;
+        properties = getNumberProperties();
+    }
+
+    private String[] getNumberProperties() {
+        final var buffer = new StringBuilder(isEven() ? "even" : "odd")
+                .append(isBuzz() ? " buzz" : "")
+                .append(isDuck() ? " duck" : "")
+                .append(isGapful() ? " gapful" : "")
+                .append(isPalindromic() ? " palindromic" : "")
+                .append(isSpy() ? " spy" : "")
+                .append(isSquare() ? " square" : "")
+                .append(isSunny() ? " sunny" : "")
+                .append(isJumping()? " jumping" : "")
+                .append(isHappy() ? " happy" : " sad");
+        return buffer.toString().split(" ");
```

Comment:
Why do you using StringBuilder to split the result?

Reply:
For performance improving.